internal/errors/redis.go

```diff
@@ -73,10 +73,7 @@ type ErrRedisNotFoundIdentity struct {
 
 // Error returns the string of ErrRedisNotFoundIdentity.error.
 func (e *ErrRedisNotFoundIdentity) Error() string {
-	if e.err != nil {
-		return e.err.Error()
```

Comment:
like kpango said, should we define an error message for `e.err` is nil?

Reply:
How about defining a new error function in errors.go like as 
```go
ErrErrorIsNil = func(name string) {
    return Errorf("error object is nil: %s". name)
}
```

and then implement in custom error struct:

```go
func (e *customError) Error() string {
    if e.err == nil {
        return ErrErrorIsNil(reflect.typeOf(e).Elem().Name()).Error()
    }
    return e.err.Error()
}
```

any ideas?

Reply:
I think we can remove the internal `err` from the field in this situation. e.g. on line 28:
Current implementation:
```golang
	// NewErrRedisNotFoundIdentity represents a function to generate an ErrRedisNotFoundIdentity error.
	NewErrRedisNotFoundIdentity = func() error {
		return &ErrRedisNotFoundIdentity{
			err: New("error redis entry not found"),
		}
	}
```

suggestion:
```golang
	// NewErrRedisNotFoundIdentity represents a function to generate an ErrRedisNotFoundIdentity error.
	NewErrRedisNotFoundIdentity = func() error {
		return  New("error redis entry not found")
	}
```

Reply:
Thx. I'd like to confirm your suggestion.
1. Your suggestion is to change the struct. Why dir you think like this?
1. have you confirmed the impact of this change?
1. you may have to consider other implements. How do you think about this?

please check these questions. @kevindiu 
@hlts2 how do you think about those discussions? and please share your idea.

Reply:
@vankichi Why I think this is because the `err` field is not using. 
It cannot access outside this package, so I think we can safely remove it.

I have checked the implementation of  `NewErrRedisNotFoundIdentity` and seems no one is using it, so I think it is safe to make this changes to it.

Consider other implementation, I think your suggestion is fine.

Reply:
I meant did you check e.g. `internal/errors/option`? @kevindiu

Reply:
yes, I think we cannot apply this on errors/option.

but in today's meeting, PO suggested to define a string type contains the default error string and then return it.
I think it is the easiest way to fix it.

Reply:
So, is it ok?
```go
func (e *customError) Error() string {
    if e.err == nil {
	e.err = New("error redis entry not found")
    }
    return e.err.Error()
}
```

Reply:
I think it should be fine.
```go
var (
   defaultCustomErrorString = "error ....."
)

func (e *customError) Error() string {
    if e.err == nil {
	return defaultCustomErrorString
    }
    return e.err.Error()
}
``

Reply:
well, so could you give me your idea about the `derfaultCustomErrorString`.?

Reply:
the idea is to return a default string when `err` is nil, without creating error type as you suggested.
I think it is the most simplest way and clean way to resolve this issue.

Reply:
sorry, what I wanted to know is that can you show me the string example that is defined as `defaultCustomErrorString`

Reply:
sorry I misunderstood your question.
it is the same as you suggested. 
the variable name and the string value will be changed according to the target error type.

for instance, for the error `ErrRedisNotFoundIdentity`, 
```go
var (
   defaultErrRedisNotFoundIdentityString = "error redis entry not found"
)
```

Reply:
@kevindiu 
It is good when you have to think about only this case.
IMO, we should define the default error string variable that can cover all custom errors.

Reply:
@vankichi thank you for your comment.
To simplify the implementation, do you think the below suggestion is good?
```go
func nilErrString(err string) string {
     return fmt.Sprintf("error object is nil: %s", err)
}
```

Reply:
@kevindiu 
Looks better!
maybe not needed the input value :thinking:

Reply:
@vankichi Sorry my suggestion is not clear. maybe we need to know which error contains nil error.
To use this function, we pass the error name to this function. e.g. 
```go
func (e *ErrRedisNotFoundIdentity) Error() string {
    if e.err == nil {
        return nilErrString("ErrRedisNotFoundIdentity")
    }
    return e.err.Error()
}
```

Reply:
the log will show the point where the error occurs, so we do not have to use any input value to specify which custom error causes the error.

Reply:
Do you mean by seeing the stack trace? 
I think sometimes the log only shows `[ERR] xxxxxx` and the cause of the error may not be visible.

Reply:
Please let me know if I misunderstood, thank you :)

Reply:
@vankichi remind :)

Reply:
how is it?
https://github.com/vdaas/vald/commit/5c7fd9cb3c6997b5e1d3ba2e91d4154bfa0cf81f

Reply:
@vankichi I think it is good.
cc: @hlts2 what do you think?